From 2a246ee8129e7cd4660fe76f7ab656191be7bc5e Mon Sep 17 00:00:00 2001
From: Jae Hyun Yoo <jae.hyun.yoo@intel.com>
Date: Thu, 11 Mar 2021 11:23:00 -0800
Subject: [PATCH] Fix build error

Signed-off-by: Jae Hyun Yoo <jae.hyun.yoo@intel.com>
---
 stdlib/canonicalize.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stdlib/canonicalize.c b/stdlib/canonicalize.c
index 698f9ede2557..cac1f73d7471 100644
--- a/stdlib/canonicalize.c
+++ b/stdlib/canonicalize.c
@@ -198,7 +198,7 @@ static char *
 realpath_stk (const char *name, char *resolved,
               struct scratch_buffer *rname_buf)
 {
-  char *dest;
+  char *dest = NULL;
   char const *start;
   char const *end;
   int num_links = 0;
-- 
2.17.1

